custom/icds_reports/sqldata/exports/growth_tracker_report.py

```diff
@@ -0,0 +1,178 @@
+
+
+class GrowthTrackerExport(ExportableMixin, IcdsSqlData):
+    title = 'Child Growth Tracking'
+    table_name = 'child_health_monthly_view'
+
+    def get_excel_data(self, location):
+
+        def _check_case_presence(case_id, column, data_dict):
+            return data_dict[case_id][column] if case_id in data_dict.keys() else "N/A"
+
+        def _fetch_data(filters, order_by, case_by_grouping=False):
+            query_set = ChildHealthMonthlyView.objects.filter(**filters).order_by(*order_by)
+            data_month = query_set.values('state_name', 'district_name', 'block_name', 'supervisor_name',
+                                          'awc_name', 'awc_site_code', 'person_name', 'dob', 'mother_name',
+                                          'mother_phone_number', 'pse_days_attended', 'lunch_count',
+                                          'current_month_nutrition_status',
+                                          current_month_wasting_column(self.beta),
+                                          current_month_stunting_column(self.beta), 'case_id')
+            if case_by_grouping is True:
+                data_month = {item['case_id']: item for item in data_month}
+            return data_month
+
+        filters = {
+            "month": self.config['month'],
+            "age_in_months__lte": 60,
+            "valid_in_month": 1
+        }
+        initial_month = self.config['month']
+
+        if self.loc_level == 5:
+            filters['awc_id'] = location
+            order_by = ('person_name',)
+        elif self.loc_level == 4:
+            filters['supervisor_id'] = location
+            order_by = ('awc_name','person_name')
+        elif self.loc_level == 3:
+            filters['block_id'] = location
+            order_by = ('supervisor_name', 'awc_name','person_name')
+        elif self.loc_level == 2:
+            filters['district_id'] = location
+            order_by = ('block_name', 'supervisor_name', 'awc_name','person_name')
+        elif self.loc_level == 1:
+            filters['state_id'] = location
+            order_by = ('district_name', 'block_name', 'supervisor_name', 'awc_name','person_name')
+        else:
+            order_by = ('state_name', 'district_name', 'block_name', 'supervisor_name', 'awc_name','person_name')
+
+        # Sample cost of each query(if data fetched for single month) is 0.98..3012.55
+        # Sample cost of query(if three months data fetched in single query) is 2.59..43594.85
+        # Sample are tested at supervisior levels
+
+        data_initial_month = _fetch_data(filters, order_by)
+        filters["month"] = initial_month - relativedelta(months=1)
+        data_past_month = _fetch_data(filters, order_by, True)
+        filters["month"] = initial_month - relativedelta(months=2)
+        data_past_month_2 = _fetch_data(filters, order_by, True)
+        filters["month"] = initial_month
+
+
+        month_1 = (initial_month - relativedelta(months=2)).strftime('%Y-%m-%d')
```

Comment:
good to have the convention the same with the above  data variables above. You could change any of the ones to match to another.

Reply:
updated

Reply:
THanks